storage/Devices/Partitionable.h

```diff
@@ -0,0 +1,58 @@
+#ifndef STORAGE_PARTITIONABLE_H
+#define STORAGE_PARTITIONABLE_H
+
+
+#include <vector>
+
+#include "storage/Devices/BlkDevice.h"
+#include "storage/Devices/PartitionTable.h"
+#include "storage/StorageInterface.h"
+
+
+namespace storage
+{
+
+    // abstract class
```

Comment:
I miss some documentation, what does it abstract? How does it differ from the generic `BlkDevice`?